contracts/src/Assets.sol

```diff
@@ -75,12 +87,19 @@ library Assets {
             } else {
                 revert Unsupported();
             }
+            // If the final destination chain is not AssetHub, then the fee needs to additionally
+            // include the cost of executing an XCM on the final destination parachain.
+            extraFee = 2 * $.sendTokenFee;
```

Comment:
Suggestion: Why not call `sendTokenFee(ParaID assetHubParaID, ParaID destinationChain)` here? Is it because we are worried about gas costs because its `external`? I think we should have an `internal` implementation and use it for both here and in `sendTokenFee(ParaID assetHubParaID, ParaID destinationChain)` so that the fee can be tuned in one place.

Reply:
yeah mostly worried about gas costs. 

Gonna have to redo this code though, see my latest comment further below.